### PR TITLE
Interpolate stops size

### PR DESCRIPTION
**Pull request by karimnaaji**
Linearly interpolate between stops size from the vertex shader at fractional zoom levels.

### CODE REVIEW

**Comment by hjanetzek** [comment]
:+1: Works! 

Since the label transformations are computed for each frame I wonder if it wouldn't make sense to just have a single screen-position attribute that represents the actual position on screen instead of computing it in the shader from screen-position + position + extrusion attributes.

**Comment by karimnaaji** [comment]
@hjanetzek , yeah I thought of this at some point; but since our overhead is mostly on the CPU right now it might be good to let this to the vertex shader. If it's for memory consumption though, it would make sense to move this out, to reduce the buffer size.

We can address that once we start reducing buffer sizes while packing attributes.

**Comment by hjanetzek** [comment]
ok then, ready to merge?

**Comment by karimnaaji** [comment]
:+1:

File: core/resources/shaders/point.vs [hunk 1]
```diff
@@ -46,10 +53,18 @@ void main() {
         float st = sin(a_rotation);
         float ct = cos(a_rotation);
 
+        vec2 vertexPos = a_position;
+        float de = a_extrude.y;
+
+        if (de != 0.0) {
+            float dz = u_map_position.z - abs(u_tile_origin.z);
+            vertexPos.xy += dz * unitExtrude[int(a_extrude.x)].xy * de;
```
Comment (karimnaaji) [inline core/resources/shaders/point.vs]: Accessing an array element with a non-constant or non-loop variable makes some drivers not receive this statement. Will follow-up on a fix related to this.

**Pull request merged**